src/eigen.jl

```diff
@@ -124,8 +129,8 @@ end
 end
 
 @inline function _eig(s::Size, A::Base.LinAlg.RealHermSymComplexHerm{T}, permute, scale) where {T <: Real}
-    eigen = eigfact(Hermitian(Array(parent(A))); permute=permute, scale=scale)
-    return (s(eigen.values), s(eigen.vectors)) # Return a SizedArray
+    eigen = eigfact(Hermitian(Array(parent(A))))
+    return (SVector{s[1], T}(eigen.values), SMatrix{s[1], s[2], T}(eigen.vectors)) # Return a SizedArray
```

Comment:
No strong opinion from me. It's hard to imagine a really *good* reason for the user of the API to care about this one way or another. Using it as an indication that some allocation has happened seems a bit of a blunt instrument to me.  It's not hard to imaging a different algorithm where sufficient postprocessing occurs that it's better to return an SVector.

Perhaps, for consistency, it would be best if we always just return something immutable. It seems bad to me that larger vs smaller sizes might return containers with different mutability.